### PR TITLE
[SPARK-11673][SQL] Remove the normal Project physical operator (and keep TungstenProject)

### PR DESCRIPTION
**Pull request by rxin**
Also make full outer join being able to produce UnsafeRows.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala [hunk 1]
```diff
@@ -176,6 +176,22 @@ case class Invoke(
   }
 }
 
+case class SerializeUsingKryo(child: Expression) extends UnaryExpression {
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala]: i need to revert this change

**Comment by rxin** [comment]
cc @davies

**Comment by rxin** [comment]
OK let's work on this one after https://github.com/apache/spark/pull/9645 is merged.

**Comment by cloud-fan** [comment]
should we also remove the `newProjection` and `newMutableProjection` in `SparkPlan`?

**Comment by rxin** [comment]
I think so, but usually I find it easier to remove stuff incrementally (due to breaking tests).

**Comment by rxin** [comment]
Oops - did anybody review this? I've already merged it. Can somebody take a closer look. Thanks.

cc @cloud-fan @davies @yhuai

**Comment by yhuai** [comment]
@cloud-fan I think we should keep `newMutableProjection` in `SparkPlan`. Or, do we have any other way to create a mutable projection?

**Comment by yhuai** [comment]
looks good to me

**Comment by davies** [comment]
LGTM

**Comment by cloud-fan** [comment]
LGTM
